File: simulador-presupuesto-personal/Frontend/src/app/models/movimiento.model.ts
/*
  Modelo Movimiento.

  Representa un ingreso o gasto registrado por el usuario.

  Este modelo es usado por:
  - el formulario para crear movimientos;
  - la lista para mostrarlos;
  - los servicios para guardarlos en LocalStorage;
  - el Web Worker para calcular el presupuesto.
*/

export type TipoMovimiento = 'ingreso' | 'gasto';

export interface Movimiento {
  id: number;
  tipo: TipoMovimiento;
  concepto: string;
  categoria: string;
  monto: number;
  fecha: string;
}

File: simulador-presupuesto-personal/Backend/models/movimiento_model.py
"""
movimiento_model.py

Define los modelos relacionados con los movimientos financieros.

Un movimiento representa un ingreso o un gasto dentro del presupuesto.
"""

from pydantic import BaseModel, Field
from typing import Literal


class MovimientoCreate(BaseModel):
    """
    Modelo usado cuando el frontend envía un nuevo movimiento.

    No incluye id porque el backend lo genera.
    No incluye fecha porque el backend la asigna al registrar el movimiento.
    """

    tipo: Literal["ingreso", "gasto"]
    concepto: str = Field(min_length=1)
    categoria: str = Field(min_length=1)
    monto: float = Field(gt=0)


class Movimiento(BaseModel):
    """
    Modelo completo de un movimiento ya registrado.

    Incluye id y fecha generados por el backend.
    """

    id: int
    tipo: Literal["ingreso", "gasto"]
    concepto: str
    categoria: str
    monto: float
    fecha: str